### PR TITLE
bpo-38693: Use f-strings instead of str.format() within importlib

### PR DESCRIPTION
**Pull request by gpshead**
This is a small performance improvement, especially for one or two hot
places such as `_handle_fromlist()` that are called a lot and the
`str.format()` method was being used just to join two strings with a `'.'`.

Otherwise it is merely a readability improvement.

This could be backported to 3.8 as it does not change any logic.

I kept `_ERR_MSG` as an attribute in `importlib._bootstrap` and `imp`
as I wasn't sure if there were other things in the world that might
refer to those.  They're private and could go away in 3.9 but should
not within 3.8 just in case.

https://bugs.python.org/issue38693

### CODE REVIEW

**Review by ericvsmith — COMMENTED** [review]
I think this is generally a good idea, and all of the changes look correct.

However, I'm not sure I'd remove the use of _ERR_MSG, since it is used outside of this file. Either that, or remove it in Lib/imp.py, too (the only other place it's used).

I'm also not sure what value is added by _ERR_MSG_PREFIX. It doesn't seem to be used anywhere. I'd just roll it into _ERR_MSG, if it's being kept.

**Comment by brettcannon** [comment]
`_ERR_MSG` was actually created specifically because there have been numerous instances where that sentence has been formatted differently across the code base (with or without a period, capitalized or not, wording, etc.). I think `_ERR_MSG_PREFIX` was created because it was used from `import.c`.

**Review by brettcannon — CHANGES_REQUESTED** [review]
Need to decide what the ultimate fate of `_ERR_MSG` is.

**Comment by csabella** [comment]
@gpshead, please take a look at the review comments.  Thanks!

**Review by jaraco — APPROVED** [review]

**Comment by ambv** [comment]
This missed the boat for inclusion in Python 3.9 which accepts security fixes only as of today.

**Comment by gpshead** [comment]
> I'm not sure I'd remove the use of _ERR_MSG, since it is used outside of this file. Either that, or remove it in Lib/imp.py, too (the only other place it's used).

I left `_ERR_MSG` and `_ERR_MSG_PREFIX` in place as they could be used by others. The code now uses the latter within its own f-string.

PTAL

File: Lib/importlib/_bootstrap.py [hunk 1]
```diff
@@ -1156,8 +1154,7 @@ def _find_and_load(name, import_):
         _lock_unlock_module(name)
 
     if module is None:
-        message = ('import of {} halted; '
-                   'None in sys.modules'.format(name))
+        message = (f'import of {name} halted; None in sys.modules')
```
Comment (ericvsmith) [inline Lib/importlib/_bootstrap.py]: The parens here can be deleted; it looks like they were previously used for the multi-line string.

**Review by ericvsmith — COMMENTED** [review]
Other than the tiny nit I mentioned, LGTM.

**Review by ericvsmith — APPROVED** [review]

**Pull request merged**